src/compile/data/parse.ts

```diff
@@ -0,0 +1,193 @@
+import {MAIN, RAW} from '../../data';
+import {Dict} from '../../util';
+import {FacetModel} from '../facet';
+import {Model} from '../model';
+import {UnitModel} from '../unit';
+import {AggregateNode} from './aggregate';
+import {BinNode} from './bin';
+import {DataFlowNode, OutputNode} from './dataflow';
+import {FacetNode} from './facet';
+import {ParseNode} from './formatparse';
+import {DataComponent} from './index';
+import {NonPositiveFilterNode} from './nonpositivefilter';
+import {NullFilterNode} from './nullfilter';
+import {OrderNode} from './pathorder';
+import {SourceNode} from './source';
+import {StackNode} from './stack';
+import {TimeUnitNode} from './timeunit';
+import {parseTransformArray} from './transforms';
+
+function parseRoot(model: Model, sources: Dict<SourceNode>): DataFlowNode {
+  if (model.data) {
+    // If we have data, try to insert it into existing sources.
```

Comment:
Do you really insert anything here? I think you just try to "reuse existing sources"